### PR TITLE
feat: map click event

### LINKED ISSUE #2502: [map] Map click event
### Motivation

The map should provide a way to listen for click events, for example to get information about the coordinates, or add markers at the clicked position.

### Acceptance criteria
- Map allows registering event listeners for click events
- Click event contains information about the clicked coordinates
- Click event contains information about the clicked 2D position, same as regular DOM events

### PR DESCRIPTION
**Pull request by hdamr**
This PR adds:

Ability to add click listener for click on the map
Adds ITs for checking click coordinates and also pixels of the clicked point.
Fixes #2502

### CODE REVIEW

File: vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/main/java/com/vaadin/flow/component/map/MapEventsPage.java [hunk 1]
```diff
@@ -34,6 +34,13 @@ public MapEventsPage() {
             eventData.setText(eventDataText);
         });
 
+        map.addMapClickEventListener(event -> {
+            eventData.setText(event.getCoordinate().getX() + ";"
+                    + event.getCoordinate().getY());
+            viewState.setText(event.getMouseDetails().getAbsoluteX() + ";"
+                    + event.getMouseDetails().getAbsoluteY());
```
Comment (sissbruecker) [inline vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/main/java/com/vaadin/flow/component/map/MapEventsPage.java:41]: The view state div has a different purpose in another test, we should not store the pixel values in it. Would suggest to just append the pixel values to the event data div.

File: vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/test/java/com/vaadin/flow/components/map/MapEventsIT.java [hunk 2]
```diff
@@ -41,8 +49,47 @@ public void changeViewPort_correctEventDataReceived() {
...
+
+        double xCoordinate = Double.parseDouble(parts[0]);
+        double yCoordinate = Double.parseDouble(parts[1]);
+
+        Assert.assertEquals(-1956787.9241005122, xCoordinate, 0.1);
+        Assert.assertEquals(1956787.9241005122, yCoordinate, 0.01);
     }
 
+    @Test
+    public void mapClick_correctPixelReceived() {
```
Comment (sissbruecker) [inline vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/test/java/com/vaadin/flow/components/map/MapEventsIT.java:81]: I think this should be part of `mapClick_correctEventDataReceived`

File: vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/test/java/com/vaadin/flow/components/map/MapEventsIT.java [hunk 3]
```diff
@@ -41,8 +49,47 @@ public void changeViewPort_correctEventDataReceived() {
...
+
+        map.clickAtCoordinates(-1956787.9241005122, 1956787.9241005122);
+
+        String[] parts = eventDataDiv.getText().split(";");
+
+        double xCoordinate = Double.parseDouble(parts[0]);
+        double yCoordinate = Double.parseDouble(parts[1]);
+
+        Assert.assertEquals(-1956787.9241005122, xCoordinate, 0.1);
+        Assert.assertEquals(1956787.9241005122, yCoordinate, 0.01);
```
Comment (sissbruecker) [inline vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/test/java/com/vaadin/flow/components/map/MapEventsIT.java:77]: Nit: different deltas, also deltas could be much smaller for coordinates

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/MapBase.java [hunk 4]
```diff
@@ -141,6 +142,11 @@ public Registration addViewMoveEndEventListener(
         return addListener(MapViewMoveEndEvent.class, listener);
     }
 
+    public Registration addMapClickEventListener(
```
Comment (sissbruecker) [inline vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/MapBase.java:145]: Missing Javadoc

Also let's remove `Map` from the name, to keep this consistent with `addViewMoveEndEventListener`

File: vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/events/MapClickEvent.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+import com.vaadin.flow.component.ComponentEvent;
+import com.vaadin.flow.component.DomEvent;
+import com.vaadin.flow.component.EventData;
+import com.vaadin.flow.component.map.MapBase;
+import com.vaadin.flow.component.map.configuration.Coordinate;
+import elemental.json.JsonArray;
+
+@DomEvent("map-click")
+public class MapClickEvent extends ComponentEvent<MapBase> {
```
Comment (sissbruecker) [inline vaadin-map-flow-parent/vaadin-map-flow/src/main/java/com/vaadin/flow/component/map/events/MapClickEvent.java:11]: Missing Javadoc on class and getters

File: vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/main/java/com/vaadin/flow/component/map/MapEventsPage.java (hunk 1 above)
Comment (hdamr) [reply vaadin-map-flow-parent/vaadin-map-flow-integration-tests/src/main/java/com/vaadin/flow/component/map/MapEventsPage.java:41]: Done.

**Review by sissbruecker — APPROVED** [review]

**Pull request merged**